### PR TITLE
Allow teleport-cluster-agent chart to use an existing volume for the data directory

### PR DESCRIPTION
**Pull request by stefansedich**
We need to be able to mount an existing persisted volume for our data directory. This PR allows the user to use another volume via `extraVolumes` and then change the volume used for the data directory mount.

It keeps the existing behavior of creating the `emptyDir` volume by default but once `existingDataVolume` is set it will not create the volume anymore and assume the user has added the volume via `extraVolumes`.

### CODE REVIEW

**Comment by benarent** [comment]
Hi @stefansedich Thanks for your first PR. I've assigned @webvictim to review this PR

**Comment by stefansedich** [comment]
@webvictim wondering if you had a chance to look over this PR, let me know if there is anything I need to change to get it merged.

**Comment by russjones** [comment]
@webvictim Can you review?

**Review by webvictim — CHANGES_REQUESTED** [review]
I think that this change itself is OK - there's another PR in #7123 which also modifies this behaviour, but I consider this 
approach to be much simpler and easier to understand.

Please add this value to the Helm chart reference at https://github.com/gravitational/teleport/blob/master/docs/pages/kubernetes-access/helm/reference.mdx so it's kept up to date.

**Comment by stefansedich** [comment]
> I think that this change itself is OK - there's another PR in #7123 which also modifies this behaviour, but I consider this
> approach to be much simpler and easier to understand.
> 
> Please add this value to the Helm chart reference at https://github.com/gravitational/teleport/blob/master/docs/pages/kubernetes-access/helm/reference.mdx so it's kept up to date.

Docs added @webvictim this should be good to go!

**Review by webvictim — APPROVED** [review]
@stefansedich Thanks - this will be merged once it has another approving review.

File: docs/pages/kubernetes-access/helm/reference.mdx [hunk 1]
```diff
@@ -1232,6 +1232,27 @@ This can be used for joining a Teleport agent to a Teleport cluster which does n
   One option might be to use Teleport's built-in [ACME support](#acme) enable [cert-manager support](#highavailabilitycertmanager).
 </Admonition>
 
+## `existingDataVolume`
+
+| Type | Default value |
+| - | - |
+| `string` | `""` |
+
+When `existingDataVolume` is set to the name of an existing volume, the `/var/lib/teleport` mount will use this volume instead of creating a new `emptyDir` volume.
```
Comment (inertial-frame) [inline docs/pages/kubernetes-access/helm/reference.mdx:1241]: I'm assuming that `if set` (below) means supplying a value other than the empty string?

Can you slightly rephrase this to better indicate that in the documentation - that the empty string is ignored? Specifically, that it will look for the directory `""` and use the default when it can't be found.

Otherwise, this is super helpful thanks!

File: examples/chart/teleport-kube-agent/templates/deployment.yaml [hunk 2]
```diff
@@ -111,7 +111,7 @@ spec:
           name: "auth-token"
           readOnly: true
         - mountPath: /var/lib/teleport
-          name: "data"
+          name: {{ default "data" .Values.existingDataVolume }}
```
Comment (inertial-frame) [inline examples/chart/teleport-kube-agent/templates/deployment.yaml:114]: Verified that no modern Linux system uses `""` for directory names (relative, system, or otherwise): https://unix.stackexchange.com/questions/160539/when-does-the-empty-string-denote-the-current-directory 

Defaults to "data" if the `existingDataVolume` value doesn't exist. 👍🏼

**Review by inertial-frame — CHANGES_REQUESTED** [review]

File: docs/pages/kubernetes-access/helm/reference.mdx (hunk 1 above)
Comment (stefansedich) [reply docs/pages/kubernetes-access/helm/reference.mdx:1241]: Not sure I follow, in Helm land it is pretty acceptable that a string is set if it is not `null` or `""`, so saying `if set` indicates it is anything but those values.

**Comment by stefansedich** [comment]
@inertial-frame what was needed now to get this PR in? I would love to move off our fork.

**Comment by webvictim** [comment]
@stefansedich He’s no longer with Teleport, this one just needs a rubber stamp again.

**Review by russjones — APPROVED** [review]
Bot.

**Comment by russjones** [comment]
@webvictim Please backport this to `branch/v7` as well, that's where we will be releasing Helm charts from soon.
